flink-core/src/test/java/org/apache/flink/util/FileUtilsTest.java

```diff
@@ -259,6 +267,67 @@ public void testCompression() throws IOException {
 		assertDirEquals(compressDir.resolve(originalDir), extractDir.resolve(originalDir));
 	}
 
+	@Test
+	public void testListFilesInPathWithoutAnyFileReturnEmptyList() throws IOException {
+		final java.nio.file.Path testDir = tmp.newFolder("_test_0").toPath();
+
+		assertTrue(Collections.<File>emptyList() ==
+			FileUtils.listFilesInPath(testDir.toFile(), f -> f.getName().endsWith(".jar")));
+	}
+
+	@Test
+	public void testListFilesInPath() throws IOException {
+		final java.nio.file.Path testDir = tmp.newFolder("_test_1").toPath();
+		final Tuple3<Collection<File>, Collection<File>, Collection<URL>> result = prepareTestFiles(testDir);
```

Comment:
I think it's better to let `prepareTestFiles` only return `Collection<File>`.
Each test case can convert it to its expected result with `toRelativeFiles` or `toRelativeURLs`.